samza-core/src/main/java/org/apache/samza/execution/LocalJobPlanner.java

```diff
@@ -119,7 +119,12 @@ private void createStreams(String planId, List<StreamSpec> intStreams, StreamMan
       return;
     }
 
-    DistributedLockWithState lockWithState = coordinationUtils.getLockWithState(planId);
+    Boolean isAppModeBatch = new ApplicationConfig(userConfig).getAppMode() == ApplicationConfig.ApplicationMode.BATCH;
+    String lockId = planId;
+    if(isAppModeBatch && runId != null) {
```

Comment:
add comment

Reply:
done. please let me know if the comment added is clear.